### PR TITLE
fix: decrease ctrlr_loss_timeout_sec for base bdev

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#9874

### CODE REVIEW

**Comment by innobead** [comment]
Build AMD64 PR check failed. 

> Downloading dapper
.dapper.tmp version v0.6.0
./.dapper ci
ERROR: permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Head "http://%2Fvar%2Frun%2Fdocker.sock/_ping": dial unix /var/run/docker.sock: connect: permission denied
time="2024-11-28T11:[5](https://github.com/longhorn/go-spdk-helper/actions/runs/12068254976/job/33653143781#step:4:6)1:35Z" level=fatal msg="exit status 1"
make: *** [Makefile:11: ci] Error 1
Error: Process completed with exit code 2.

**Comment by derekbit** [comment]
> Build AMD64 PR check failed.
> 
> > Downloading dapper
> > .dapper.tmp version v0.6.0
> > ./.dapper ci
> > ERROR: permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Head "http://%2Fvar%2Frun%2Fdocker.sock/_ping": dial unix /var/run/docker.sock: connect: permission denied
> > time="2024-11-28T11:[5](https://github.com/longhorn/go-spdk-helper/actions/runs/12068254976/job/33653143781#step:4:6)1:35Z" level=fatal msg="exit status 1"
> > make: *** [Makefile:11: ci] Error 1
> > Error: Process completed with exit code 2.

Yes, environmental issue. Waiting for @yangchiu fix it.

**Review by innobead — CHANGES_REQUESTED** [review]
`go-spdk-helper` binary accidentally added?

File: pkg/types/types.go [hunk 1]
```diff
@@ -23,9 +23,15 @@ const (
 	ShallowCopyStateError      = "error"
 
 	DefaultCtrlrLossTimeoutSec = 30
-	// DefaultReconnectDelaySec can't be more than DefaultFastIoFailTimeoutSec, same for non-default values.
+	// DefaultReplicaReconnectDelaySec can't be more than DefaultReplicaFastIOFailTimeoutSec, same for non-default values.
```
[the remark this answers is not in the document]
Comment (derekbit) [reply pkg/types/types.go]: Fixed. Thank you.

File: pkg/spdk/client/basic.go [hunk 2]
```diff
@@ -770,22 +770,13 @@ func (c *Client) BdevNvmeGetControllers(name string) (controllerInfoList []spdkt
 // Parameters, ctrlr_loss_timeout_sec, reconnect_delay_sec, and fast_io_fail_timeout_sec, are
 // for I/O error resiliency. They can be overridden if they are given by the RPC bdev_nvme_attach_controller.
 //
-// "ctrlrLossTimeoutSec": Controller loss timeout in seconds
```
[the remark this answers is not in the document]
Comment (derekbit) [reply pkg/spdk/client/basic.go]: I added them back. 
The parameters are for global settings.
We can override the timeouts by parameters of bdev_nvme_attach_controller, so I think keeping them is fine.

File: pkg/types/types.go [hunk 3]
```diff
@@ -23,9 +23,15 @@ const (
 	ShallowCopyStateError      = "error"
 
 	DefaultCtrlrLossTimeoutSec = 30
-	// DefaultReconnectDelaySec can't be more than DefaultFastIoFailTimeoutSec, same for non-default values.
+	// DefaultReconnectDelaySec can't be more than DefaultFastIOFailTimeoutSec.
 	DefaultReconnectDelaySec    = 2
 	DefaultFastIOFailTimeoutSec = 15
+
+	DefaultReplicaCtrlrLossTimeoutSec = 15
```
Comment (DamiaSan) [inline pkg/types/types.go:30]: Maybe we can put a brief description with the difference between `DefaultCtrlrLossTimeoutSec` and `DefaultReplicaCtrlrLossTimeoutSec` .

Comment (derekbit) [reply pkg/types/types.go]: Removed DefaultReplicaCtrlrLossTimeoutSec and just changed the value of DefaultCtrlrLossTimeoutSec and DefaultFastIOFailTimeoutSec. The two values are for CLI's parameters, and we won't use them for APIs.

**Review by DamiaSan — APPROVED** [review]

**Pull request merged**